pkg/broker/eventutil/hops.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventutil
+
+import (
+	cetypes "github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"go.uber.org/zap"
+)
+
+const (
+	hopsAttribute = "kngcpbrokerhops"
+)
+
+// Hops manages event hops.
+type Hops struct {
+	Logger *zap.Logger
```

Comment:
I recommend having `ctx` passed in to the methods and using `logging.GetFromContext(ctx)` rather than storing it in the struct. That way any additional information about the event being processed is retained.